diff --git a/CMakeLists.txt b/CMakeLists.txt
index aa094b1..0218fa7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,17 +47,17 @@ include_directories(
 
 add_executable(sllidar_node	src/sllidar_node.cpp ${SLLIDAR_SDK_SRC})
 # target_link_libraries(sllidar_node ${ament_cmake_LIBRARIES})
-ament_target_dependencies(sllidar_node
-  rclcpp
-  std_srvs
-  sensor_msgs
+target_link_libraries(sllidar_node
+  rclcpp::rclcpp
+  ${std_srvs_TARGETS}
+  ${sensor_msgs_TARGETS}
 )
 
 add_executable(sllidar_client src/sllidar_client.cpp)
-ament_target_dependencies(sllidar_client
-  rclcpp
-  std_srvs
-  sensor_msgs
+target_link_libraries(sllidar_client
+  rclcpp::rclcpp
+  ${std_srvs_TARGETS}
+  ${sensor_msgs_TARGETS}
 )
 ################################################################################
 # Install
